RoddyCore/src/de/dkfz/roddy/execution/jobs/Job.groovy

```diff
@@ -333,6 +338,100 @@ class Job extends BEJob<BEJob, JobResult> {
         path
     }
 
+
+    /**
+     * Stores a new job jobState info to an execution contexts job jobState log file.
+     *
+     * @param job
+     */
+    void appendToJobStateLogfile(BatchEuphoriaJobManager jobManager, ExecutionContext executionContext, BEJobResult res, OutputStream out = null) {
+        // Ugly, but works. Java/Groovy's missing runtime dispatch on method arguments sucks.
+        if (jobManager instanceof ClusterJobManager) {
+            appendToJobStateLogfileForCluster(executionContext, res)
```

Comment:
Neither do I. It is because Groovy does not support dynamic dispatch on more than the object itself (a.k.a. multiple dispatch). But I agree with your previous comment (!) that this code does not belong into ExecutionService. Conceptually, this code belongs here. 

I now better understand the other instanceOf cases that we have. They are there due to the missing features in the object system (or of object systems in general).